common/utils/index.js

```diff
@@ -745,17 +747,25 @@ function pushServicePlanToApiServer() {
   }
 }
 
-function registerInterOperatorCrds() {
+function registerSFEventsCrd() {
   const eventmesh = require('../../data-access-layer/eventmesh');
   return Promise.all([
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICES),
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_PLANS),
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEINSTANCES),
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS),
     eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INSTANCE, CONST.APISERVER.RESOURCE_TYPES.SFEVENT)
   ]);
 }
 
+function waitWhileCRDsAreRegistered() {
+  // We assume here that the Helm chart based deployment or the pre-start script of interoperator job in case of BOSH based deployment will take care of registering the CRDs. While it is not registered, we wait!
+  const eventmesh = require('../../data-access-layer/eventmesh');
+  logger.info('Checking if the CRDs are already registered');
+  return eventmesh.apiServerClient.getCustomResourceDefinition(CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICES + '.' + CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR)
+    .then(() => eventmesh.apiServerClient.getCustomResourceDefinition(CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_PLANS + '.' + CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR))
+    .catch(NotFound, () => {
+      logger.info('Waiting for the CRDs to get registered');
+      return Promise.delay(1000).then(() => waitWhileCRDsAreRegistered());
```

Comment:
Please put `1000` in constants file. Can be placed under `constants.APISERVER` .

Reply:
Done